packages/streaming-image-volume-loader/src/BaseStreamingImageVolume.ts

```diff
@@ -204,16 +223,214 @@ export default class BaseStreamingImageVolume extends ImageVolume {
     this.loadStatus.callbacks = [];
   }
 
+  protected callLoadStatusCallback(evt) {
+    const { framesUpdated, framesProcessed, totalNumFrames } = evt;
+    const { volumeId, reRenderFraction, loadStatus, metadata } = this;
+    const { FrameOfReferenceUID } = metadata;
+
+    // TODO: probably don't want this here
+    if (this.autoRenderOnLoad) {
+      if (
+        framesUpdated > this.reRenderTarget ||
+        framesProcessed === totalNumFrames
+      ) {
+        this.reRenderTarget += reRenderFraction;
+        autoLoad(volumeId);
+      }
+    }
+    if (framesProcessed === totalNumFrames) {
+      loadStatus.callbacks.forEach((callback) => callback(evt));
+
+      const eventDetail = {
+        FrameOfReferenceUID,
+        volumeId: volumeId,
+      };
+
+      triggerEvent(
+        eventTarget,
+        Enums.Events.IMAGE_VOLUME_LOADING_COMPLETED,
+        eventDetail
+      );
+    }
+  }
+
+  protected updateTextureAndTriggerEvents(
+    volume: BaseStreamingImageVolume,
+    imageIdIndex,
+    imageId,
+    status = ImageQualityStatus.FULL_RESOLUTION
+  ) {
+    const frameIndex = this._imageIdIndexToFrameIndex(imageIdIndex);
+    const { cachedFrames, numFrames, totalNumFrames } = this;
+    const { FrameOfReferenceUID } = this.metadata;
+    const currentStatus = cachedFrames[frameIndex];
+    if (currentStatus > status) {
+      // This is common for initial versus decimated images.
+      return;
+    }
+
+    if (cachedFrames[frameIndex] === ImageQualityStatus.FULL_RESOLUTION) {
+      return;
+    }
```

Comment:
I'm surprised about these if statements. We never returned in the old code when we had cached frames. Is this intentional?